### PR TITLE
Fixing metric computation for old vs new ranks with ties

### PR DESCRIPTION
**Pull request by lastmansleeping**

### CODE REVIEW

**Review by mohazahran — CHANGES_REQUESTED** [review]
left a minor comment.

File: python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py [hunk 1]
```diff
@@ -162,16 +126,25 @@ def get_grouped_stats(
             # we cannot compute old NDCG
             df[RankingConstants.OLD_NDCG] = 0.0
 
-    df_clicked = df[df[artificial_click_col] == 1.0]
-    df = df[df[query_key_col].isin(df_clicked[query_key_col])]
+    # Find the records with the highest relevance grade from the label column
+    df[RankingConstants.PROXY_CLICK] = df.groupby(query_key_col, group_keys=False)[label_col].apply(lambda x: x == x.max()).astype(float)
+    df_clicked = df[df[RankingConstants.PROXY_CLICK] == 1.0]
```
[the remark this answers is not in the document]
Comment (lastmansleeping) [reply python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py:131]: Sorry, missed this. Can you explain what you mean? I don't get it.

File: python/ml4ir/applications/ranking/model/ranking_model.py [hunk 2]
```diff
@@ -140,6 +140,13 @@ def evaluate(
                 ]
             )
 
+            try:
```
[the remark this answers is not in the document]
Comment (mohazahran) [reply python/ml4ir/applications/ranking/model/ranking_model.py:143]: as discussed in slack, there is an option to keep queries with no signals, so these queries would get a proxy click and may disturb the metrics and should be removed. I'm mainly bringing this up as the removed function to get the top graded relevance record used to filter out these queries.

File: python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mohazahran) [reply python/ml4ir/applications/ranking/model/metrics/helpers/metrics_helper.py:131]: s discussed in slack, there is an option to keep queries with no signals, so these queries would get a proxy click and may disturb the metrics and should be removed. I'm mainly bringing this up as the removed function to get the top graded relevance record used to filter out these queries.

**Review by mohazahran — APPROVED** [review]
looks good.

**Pull request merged**